File: ПМИВС/lab4/Lab_4_GuessNumber_Shibitov_Nikolay/app/src/main/java/com/example/lab_4_guessnumber_shibitov_nikolay/MainActivity.java
package com.example.lab_4_guessnumber_shibitov_nikolay;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

public class MainActivity extends AppCompatActivity {

    private GameViewModel viewModel;
    private EditText etGuess;
    private TextView tvResult;
    private TextView tvAttempts;
    private Button   btnGuess;
    private Button   btnNewGame;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etGuess    = findViewById(R.id.et_guess);
        tvResult   = findViewById(R.id.tv_result);
        tvAttempts = findViewById(R.id.tv_attempts);
        btnGuess   = findViewById(R.id.btn_guess);
        btnNewGame = findViewById(R.id.btn_new_game);

        viewModel = new ViewModelProvider(this).get(GameViewModel.class);

        viewModel.getResultText().observe(this,   text -> tvResult.setText(text));
        viewModel.getAttemptsText().observe(this, text -> tvAttempts.setText(text));
        viewModel.getIsGameOver().observe(this, gameOver -> {
            btnGuess.setEnabled(!gameOver);
            btnNewGame.setEnabled(gameOver);
        });

        btnGuess.setOnClickListener(v -> {
            String input = etGuess.getText().toString();
            viewModel.makeGuess(input);
            etGuess.setText("");
        });

        btnNewGame.setOnClickListener(v -> {
            viewModel.startNewGame();
            etGuess.setText("");
        });
    }
}

File: ПМИВС/lab6/Lab6_Task75_SQLite_Upgrade_Shibitov/app/src/main/java/com/example/lab6_task75_shibitov/MainActivity.kt
package com.example.lab6_task75_shibitov

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var dbHelper: BookDbHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        dbHelper = BookDbHelper(this)
        dbHelper.initData()

        findViewById<TextView>(R.id.tvStatus).text = getString(R.string.db_init_done)

        findViewById<Button>(R.id.btnView).setOnClickListener {
            startActivity(Intent(this, BooksListActivity::class.java))
        }

        findViewById<Button>(R.id.btnAdd).setOnClickListener {
            startActivity(Intent(this, AddBookActivity::class.java))
        }

        findViewById<Button>(R.id.btnEdit).setOnClickListener {
            val first = dbHelper.getFirstBook()
            if (first != null) {
                startActivity(Intent(this, EditBookActivity::class.java).apply {
                    putExtra("book_id", first.id)
                    putExtra("book_author", first.author)
                    putExtra("book_title", first.title)
                    putExtra("book_year", first.year)
                    putExtra("book_publisher", first.publisher)
                    putExtra("book_pages", first.pages)
                    putExtra("book_price", first.price)
                })
            } else {
                Toast.makeText(this, "Нет записей для редактирования", Toast.LENGTH_SHORT).show()
            }
        }
    }
}
